In [1]:
import os

In [2]:
dir_name_xml = 'images_items_annotation'
dir_name_image = 'images_items'
list_xml_file_name = os.listdir(dir_name_xml)
list_image_file_name = os.listdir(dir_name_image)

In [3]:
list_xml_file_name[:10]

['010726.xml',
 '56215-A_IMB.xml',
 '52737-A_IMT.xml',
 '65256-A_IWT.xml',
 '014871.xml',
 '011455.xml',
 '64159-A_IWT.xml',
 '35655-A_IMB.xml',
 '010837.xml',
 '013858.xml']

In [4]:
for file_name in list_xml_file_name:
    if len(file_name) > len('010726.xml'):
        os.remove(dir_name_xml + '/' + file_name)

In [5]:
len(list_xml_file_name)

10162

In [8]:
list_xml_file_name = os.listdir(dir_name_xml)
len(list_xml_file_name)

5081

5081 means the number of all items.

There are no wrongly deleted files.

In [9]:
len(list_xml_file_name) == len(list_image_file_name)

True